modules/ztp-checking-du-cluster-config.adoc

```diff
@@ -0,0 +1,518 @@
+
+:_module-type: PROCEDURE
+[id="ztp-checking-du-cluster-config_{context}"]
+= Checking that the recommended cluster configurations are applied
+
+You can check that clusters are running the correct configuration. The following procedure describes how to check the various configurations that you require to deploy a DU application in {product-title} {product-version} clusters.
+
+.Prerequisites
+
+* Deploy a cluster and tune it for vDU workloads.
+
+* Install the OpenShift CLI (`oc`).
+
+* Log in as a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Check that the default Operator Hub sources are disabled. Run the following command:
++
+[source,terminal]
+----
+$ oc get operatorhub cluster -o yaml
+----
++
+.Example output
+[source,yaml]
+----
+spec:
+    disableAllDefaultSources: true
+----
+
+. Check that all required `CatalogSource` resources are annotated for workload partitioning (`PreferredDuringScheduling`), for example:
++
+[source,terminal]
+----
+$ oc get catalogsource -A -o jsonpath='{range .items[*]}{.metadata.name}{" -- "}{.metadata.annotations.target\.workload\.openshift\.io/management}{"\n"}{end}'
+----
++
+.Example output
+[source,terminal]
+----
+certified-operators -- {"effect": "PreferredDuringScheduling"}
+community-operators -- {"effect": "PreferredDuringScheduling"}
+ran-operators -- <1>
+redhat-marketplace -- {"effect": "PreferredDuringScheduling"}
+redhat-operators -- {"effect": "PreferredDuringScheduling"}
+----
+<1> `CatalogSource` resources that are not annotated are also returned. In this example, the `ran-operators` `CatalogSource` resource is not annotated and does not have the `PreferredDuringScheduling` annotation.
++
+[NOTE]
+====
+In a properly configured vDU cluster, only a single annotated catalog source is listed.
+====
+
+. Check that all applicable {product-title} Operator namespaces are annotated for workload partitioning. This includes all Operators installed with core {product-title} and the set of additional Operators included in the reference DU tuning configuration.
++
+[source,terminal]
+----
+$ oc get namespaces -A -o jsonpath='{range .items[*]}{.metadata.name}{" -- "}{.metadata.annotations.workload\.openshift\.io/allowed}{"\n"}{end}'
+----
++
+.Example output
+[source,terminal]
+----
+default --
+openshift-apiserver -- management
+openshift-apiserver-operator -- management
+openshift-authentication -- management
+openshift-authentication-operator -- management
+----
++
+[IMPORTANT]
+====
+Additional Operators must not be annotated for workload partitioning. In the output from the previous command, additional Operators should be listed without any value on the right-hand side of the `--` separator.
+====
+
+. Check that the `ClusterLogging` configuration is correct.
+
+.. Validate that the curation schedule is appropriate for your application by running the following command:
++
+[source,terminal]
+----
+$ oc get -n openshift-logging ClusterLogging
```

Comment:
Here is an example of input/output:

```
$ oc get -n openshift-logging clusterloggings.logging.openshift.io instance -o yaml
apiVersion: logging.openshift.io/v1
kind: ClusterLogging
metadata:
  creationTimestamp: "2022-07-07T18:22:56Z"
  generation: 1
  name: instance
  namespace: openshift-logging
  resourceVersion: "235796"
  uid: ef67b9b8-0e65-4a10-88ff-ec06922ea796
spec:
  collection:
    logs:
      fluentd: {}
      type: fluentd
  curation:
    curator:
      schedule: 30 3 * * *
    type: curator
  managementState: Managed
status:
  clusterConditions:
  - lastTransitionTime: "2022-07-07T18:23:17Z"
    status: "False"
    type: CollectorDeadEnd
  - lastTransitionTime: "2022-07-07T18:23:04Z"
    message: curator is deprecated in favor of defining retention policy
    reason: ResourceDeprecated
    status: "True"
    type: CuratorRemoved
  collection:
    logs:
      fluentdStatus:
        daemonSet: collector
        nodes:
          collector-9nggh: cnfde4.ptp.lab.eng.bos.redhat.com
        pods:
          failed: []
          notReady: []
          ready:
          - collector-9nggh
  curation: {}
  logStore: {}
  visualization: {}
```

Reply:
@imiller0 please see status.clusterConditions. Looks like our recommended config may be deprecated?

Reply:
I've opened BZ 2109716 to cover the deprecated api (thank you @yliu127!).